src/apis/watch.ts

```diff
@@ -248,6 +250,21 @@ function createWatcher(
   // effect watch
   if (cb === null) {
     let running = false
+    const { Dep } = getVueInternalClasses()
+    Dep.prototype.notify = function notify() {
```

Comment:
We should avoid patching `prototype` methods.

 This patching of the prototype will run every time we call a watch, because of the scoped `running` it might cause problems.